ddsrouter_core/src/cpp/participant/implementations/rtps/DiscoveryServerParticipant.cpp

```diff
@@ -91,16 +91,40 @@ DiscoveryServerParticipant::participant_attributes_(
             {
                 has_listening_tcp_ipv4 = true;
 
-                std::shared_ptr<eprosima::fastdds::rtps::TCPv4TransportDescriptor> descriptor =
-                        std::make_shared<eprosima::fastdds::rtps::TCPv4TransportDescriptor>();
+                std::shared_ptr<eprosima::fastdds::rtps::TCPv4TransportDescriptor> descriptor;
+                bool same_wan_addr = false;
 
-                descriptor->add_listener_port(address.port());
-                descriptor->set_WAN_address(address.ip());
+                auto it = params.userTransports.begin();
+                while (it != params.userTransports.end())
+                {
+                    std::shared_ptr<eprosima::fastdds::rtps::TCPv4TransportDescriptor> tmp_descriptor =
+                            std::dynamic_pointer_cast<eprosima::fastdds::rtps::TCPv4TransportDescriptor>(*it);
+
+                    if ((tmp_descriptor != nullptr) && (address.ip() == tmp_descriptor->get_WAN_address()))
+                    {
+                        descriptor = tmp_descriptor;
+                        same_wan_addr = true;
+                        params.userTransports.erase(it);
+                        break;
+                    }
+                }
 
-                // Enable TLS
-                if (tls_config.is_active())
+                if (same_wan_addr)
                 {
-                    tls_config.enable_tls(descriptor);
+                    descriptor->add_listener_port(address.port());
+                }
+                else
+                {
+                    descriptor = std::make_shared<eprosima::fastdds::rtps::TCPv4TransportDescriptor>();
```

Comment:
be careful, this line is not correct, it will create a new descriptor, fill it and destroy it

Reply:
The descriptor is nullptr as it was created some lines above so this comment does not apply anymore.